core/src/subgraph/provider.rs

```diff
@@ -137,3 +163,48 @@ impl<L, S> EventProducer<SubgraphProviderEvent> for SubgraphProvider<L, S> {
             .map(|s| Box::new(s) as Box<Stream<Item = SubgraphProviderEvent, Error = ()> + Send>)
     }
 }
+
+fn attribute_indexing_operations(
+    subgraph_id: SubgraphId,
+    document: Document,
+) -> Vec<AttributeIndexOperation> {
+    let mut indexing_ops = vec![];
+    for (entity_number, schema_type) in document.definitions.clone().into_iter().enumerate() {
+        if let Definition::TypeDefinition(definition) = schema_type {
+            if let TypeDefinition::Object(schema_object) = definition {
+                for (attribute_number, entity_field) in schema_object.fields.into_iter().enumerate()
+                {
+                    let field_value_type =
+                        match sast::get_valid_value_type(&entity_field.field_type) {
+                            Ok(value_type) => value_type,
+                            Err(_) => continue,
+                        };
+                    let (index_type, index_operator) = match field_value_type {
+                        ValueType::Boolean
+                        | ValueType::BigInt
+                        | ValueType::Bytes
+                        | ValueType::Float
+                        | ValueType::ID
+                        | ValueType::Int => (String::from("btree"), String::from("")),
+                        ValueType::String => (String::from("gin"), String::from("gin_trgm_ops")),
+                    };
```

Comment:
The choice of index type is tied to the storage backend so it belongs in the store implementation. How about replacing `index_type` and `index_operator` with `field_value_type` in  `AttributeIndexOperation`, and moving this match into the store?